Debug impl for Architecture is broken for ARM

STR
===

`cargo build --target=arm-unknown-linux-gnueabi`

It doesn't build because we don't specify the right ARM sub-architecture.

```
error[E0425]: cannot find value `Arm` in this scope
  --> /home/ben/code/target-lexicon/target/arm-unknown-linux-gnueabi/debug/build/target-lexicon-4ee7e8d1871c8170/out/host.rs:49:45
   |
49 |             architecture: Architecture::Arm(Arm),
   |                                             ^^^ not found in this scope
help: possible candidates are found in other modules, you can import them into scope
   |
1  | use crate::targets::Architecture::Arm;
   |
1  | use crate::targets::ArmArchitecture::Arm;
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -25,7 +25,13 @@ before_script:
     # Sometimes the component isn't actually ready after being installed, and
     # rustup update makes it ready.
     - rustup update
+    # Add targets for testing cross-compilation.
+    - rustup target add arm-unknown-linux-gnueabi
+    - rustup target add aarch64-unknown-linux-gnu
     - rustfmt +stable --version
-script: cargo test
+script:
+    - cargo test
+    - cargo build --target arm-unknown-linux-gnueabi
+    - cargo build --target aarch64-unknown-linux-gnu
 cache:
     cargo: true

--- a/build.rs
+++ b/build.rs
@@ -121,6 +121,15 @@ fn main() {
 }
 
 fn write_host_rs(mut out: File, triple: Triple) -> io::Result<()> {
+    // The generated Debug implementation for the inner architecture variants
+    // doesn't print the enum name qualifier, so import them here. There
+    // shouldn't be any conflicts because these enums all share a namespace
+    // in the triple string format.
+    writeln!(out, "#[allow(unused_imports)]")?;
+    writeln!(out, "use crate::Aarch64Architecture::*;")?;
+    writeln!(out, "#[allow(unused_imports)]")?;
+    writeln!(out, "use crate::ArmArchitecture::*;")?;
+    writeln!(out)?;
     writeln!(out, "/// The `Triple` of the current host.")?;
     writeln!(out, "pub const HOST: Triple = Triple {{")?;
     writeln!(

